`copy_from()` crashes for fields of struct types

The following code crashes
```py
@ti.dataclass
class C:
    i: int
    f: float

cf1 = C.field(shape = (2,))
cf2 = C.field(shape = (2,))

cf1.copy_from(cf2)
```
with the error message
```
TypeError: Cannot copy from a non-field object
```


Interestingly, if the copy is performed on 2D fields, the crash has a different error message:
```py
@ti.dataclass
class C:
    i: int
    f: float

cf1 = C.field(shape = (2,2))
cf2 = C.field(shape = (2,2))

cf1.copy_from(cf2)
```
```
AssertionError: Slicing is not supported on ti.field
```

--- a/python/taichi/lang/struct.py
+++ b/python/taichi/lang/struct.py
@@ -538,7 +538,7 @@ class StructField(Field):
         assert isinstance(other, Field)
         assert set(self.keys) == set(other.keys)
         for k in self.keys:
-            self.field_dict[k].copy_from(other[k])
+            self.field_dict[k].copy_from(other.get_member_field(k))
 
     @python_scope
     def fill(self, val):

--- a/tests/python/test_copy_from.py
+++ b/tests/python/test_copy_from.py
@@ -23,3 +23,32 @@ def test_scalar():
     assert y[0] == 1
     assert y[1] == 0
     assert y[2] == 3
+
+
+@test_utils.test()
+def test_struct():
+    @ti.dataclass
+    class C:
+        i: int
+        f: float
+
+    n = 16
+
+    x = C.field(shape=n)
+    y = C.field(shape=n)
+
+    x[1].i = 2
+    x[2].i = 4
+
+    y[0].f = 1.0
+    y[2].i = 3
+
+    x.copy_from(y)
+
+    assert x[0].f == 1.0
+    assert x[1].i == 0
+    assert x[2].i == 3
+
+    assert y[0].f == 1.0
+    assert y[1].i == 0
+    assert y[2].i == 3

